packages/api/src/external/shared/patient-demographics.ts

```diff
@@ -0,0 +1,211 @@
+import { Patient, splitName, splitDob } from "@metriport/core/domain/patient";
+import { Address } from "@metriport/core/domain/address";
+import { Contact } from "@metriport/core/domain/contact";
+import { USState } from "@metriport/core/domain/geographic-locations";
+
+export const addressSeparator = "||";
+
+type LinkDemoDataAddress = {
+  line?: string;
+  city?: string;
+  state?: string;
+  zip?: string;
+  country?: string;
+};
+
+export type LinkDemoData = {
+  dob?: string;
+  gender?: string;
+  names: {
+    firstName: string;
+    lastName: string;
+  }[];
+  telephoneNumbers: string[];
+  emails: string[];
+  addresses: LinkDemoDataAddress[];
+  driversLicenses: {
+    value: string;
+    state: string;
+  }[];
+  ssns: string[];
+};
+
+export function scoreLink(
+  patientDemographics: LinkDemoData,
+  linkDemographics: LinkDemoData
+): boolean {
+  let scoreThreshold = 20;
+  let score = 0;
+  // DOB exact match
+  if (patientDemographics.dob === linkDemographics.dob) score += 8;
+  // DOB approximate match
+  if (patientDemographics.dob && linkDemographics.dob) {
+    const patientDobSplit = splitDob(patientDemographics.dob);
+    const linkedDobSplit = splitDob(linkDemographics.dob);
+    if (linkedDobSplit.filter(patientDobSplit.includes).length > 2) score += 2;
+  }
+  // Gender exact match
+  if (patientDemographics.gender && linkDemographics.gender) {
+    if (patientDemographics.gender === linkDemographics.gender) score += 1;
+  }
+  // Names exact match
+  if (linkDemographics.names.some(patientDemographics.names.includes)) score += 10;
+  // Names approximate match
+  // TODO
+  // Address exact match
+  // TODO Mark the first address as current in Dash
+  const overlappingAddresses = linkDemographics.addresses.filter(a =>
+    patientDemographics.addresses.includes(a)
+  );
+  score += 2 * overlappingAddresses.length;
+  const currentPatientAddress = patientDemographics.addresses[0];
+  if (currentPatientAddress) {
+    if (linkDemographics.addresses.map(a => a.city).includes(currentPatientAddress.city))
+      score += 0.5;
+    if (linkDemographics.addresses.map(a => a.zip).includes(currentPatientAddress.zip))
+      score += 0.5;
+  }
+  // Address approximate match
+  // TODO
+  // Telephone exact match
+  const overlappingTelephone = linkDemographics.telephoneNumbers.filter(a =>
+    patientDemographics.telephoneNumbers.includes(a)
+  );
+  score += 2 * overlappingTelephone.length;
```

Comment:
in epic algo:

"Phone numbers are
cross matched with all
known phone numbers
for the patient. A
maximum of 2 points is
awarded for a match."

This is awarding more than a max of two. 2* n